### PR TITLE
layers/transformer,  layers/attention

### PR DESCRIPTION
**Pull request by xinliupitt**
# Description

> :memo: Changes in transformer and attention layer
>  
> * This change is to  to help with performance of Seq2SeqTransformer model.
> * Arguments in layer class `__init__` are added to meet different needs of using layers.

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [x] TensorFlow 2 migration

## Tests

> :memo: Please describe the tests that you ran to verify your changes.
>  
> * This change passes layers/transformer_test.py

**Test Configuration**:

## Checklist

- [ ] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [ ] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [ ] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [ ] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: official/nlp/modeling/layers/transformer.py [hunk 1]
```diff
@@ -49,6 +49,10 @@ class Transformer(tf.keras.layers.Layer):
     activity_regularizer: Regularizer for dense layer activity.
     kernel_constraint: Constraint for dense layer kernels.
     bias_constraint: Constraint for dense layer kernels.
+    use_bias: Whether to enable use_bias in attention layer.
+    norm_first: Whether to normalize inputs to attention and intermediate dense
+      layers.
+    norm_epsilon: Epsilon value to initialize normalization layers.
```
Comment (saberkun) [inline official/nlp/modeling/layers/transformer.py:55]: If it is false, what's the behavior. Please help document it.

File: official/nlp/modeling/layers/transformer.py [hunk 2]
```diff
@@ -81,6 +88,9 @@ def __init__(self,
     self._activity_regularizer = tf.keras.regularizers.get(activity_regularizer)
     self._kernel_constraint = tf.keras.constraints.get(kernel_constraint)
     self._bias_constraint = tf.keras.constraints.get(bias_constraint)
+    self._use_bias = use_bias
```
Comment (saberkun) [inline official/nlp/modeling/layers/transformer.py:91]: We need to add these new arguments to get_config?

File: official/nlp/modeling/layers/transformer.py [hunk 3]
```diff
@@ -289,6 +320,9 @@ def __init__(self,
...
```
Comment (saberkun) [inline official/nlp/modeling/layers/transformer.py:323]: get_config()?

File: official/nlp/modeling/layers/transformer_test.py [hunk 4]
```diff
@@ -251,6 +289,40 @@ def test_decoder_block_with_cache(self):
     self.assertEqual(output.shape, (2, 4, hidden_size))
     self.assertEqual(cache['value'].shape, (2, 4, 2, 8))
 
+  def test_use_bias(self):
```
Comment (saberkun) [inline official/nlp/modeling/layers/transformer_test.py:292]: use_bias_norm_first?

Comment (xinliupitt) [reply official/nlp/modeling/layers/transformer_test.py:292]: Let me do it now

**Pull request merged**